pandas/_libs/tslibs/period.pyx

```diff
@@ -1581,7 +1568,7 @@ cdef class _Period(object):
         """
         Fast creation from an ordinal and freq that are already validated!
         """
-        if ordinal == iNaT:
+        if ordinal == NPY_NAT:
```

Comment:
Looking at the output of `cython -a ...`, comparing against `iNaT` has python lookup overhead that NPY_NAT does not.